lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java

```diff
@@ -456,18 +458,27 @@ public void onConnectionStateChanged(ConnectionStateChange state) {
 			connectionWaiter.waitFor(ConnectionState.connected);
 			final String firstConnectionId = ably.connection.id;
 
-			/* Wait for ttl + idle timer to pass then disconnect */
-			try {
-				Thread.sleep(intervalBeforeReconnecting);
-			} catch(InterruptedException e) {}
-			ably.connection.connectionManager.requestState(ConnectionState.disconnected);
-
-			/* Since newTtl + newIdleInterval has passed we expect the connection to go into a suspended state */
-			connectionWaiter.waitFor(ConnectionState.suspended);
+			ably.connection.once(ConnectionEvent.disconnected, new ConnectionStateListener() {
+				@Override
+				public void onConnectionStateChanged(ConnectionStateChange state) {
+					synchronized (ably.connection.connectionManager) {
+						try {
+							/* The client will try to reconnect right away after disconnection.
+							 * We want it to stay into a disconnected state long enough
+							 * so that the connection becomes stale.
+							 */
+							ably.connection.connectionManager.wait(waitInDisconnectedState);
```

Comment:
I don't understand what the aim of this `wait()` is for. Won't this code be blocking the connectionmanager thread? Was that the intention?